{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "azdata_cell_guid": "26037d32-2047-4157-81ef-595916bd66a0"
   },
   "source": [
    "# Checkpoint Three: Cleaning Data\n",
    "\n",
    "Now you are ready to clean your data. Before starting coding, provide the link to your dataset below.\n",
    "\n",
    "My dataset: https://www.kaggle.com/bbhatt001/human-microbiome-project\n",
    "\n",
    "Import the necessary libraries and create your dataframe(s)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "azdata_cell_guid": "e8adef8e-d0f2-4640-a179-5997f11e82ca"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 2915 entries, 0 to 2914\n",
      "Data columns (total 17 columns):\n",
      " #   Column                   Non-Null Count  Dtype \n",
      "---  ------                   --------------  ----- \n",
      " 0   HMP ID                   2915 non-null   int64 \n",
      " 1   GOLD ID                  1783 non-null   object\n",
      " 2   Organism Name            2915 non-null   object\n",
      " 3   Domain                   2712 non-null   object\n",
      " 4   NCBI Superkingdom        2751 non-null   object\n",
      " 5   HMP Isolation Body Site  2915 non-null   object\n",
      " 6   Project Status           2915 non-null   object\n",
      " 7   Current Finishing Level  1579 non-null   object\n",
      " 8   NCBI Submission Status   2915 non-null   object\n",
      " 9   NCBI Project ID          2915 non-null   int64 \n",
      " 10  Genbank ID               1579 non-null   object\n",
      " 11  Gene Count               2915 non-null   int64 \n",
      " 12  IMG/HMP ID               2915 non-null   int64 \n",
      " 13  HOMD ID                  397 non-null    object\n",
      " 14  Sequencing Center        2911 non-null   object\n",
      " 15  Funding Source           2915 non-null   object\n",
      " 16  Strain Repository ID     1377 non-null   object\n",
      "dtypes: int64(4), object(13)\n",
      "memory usage: 387.3+ KB\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "hmp_df = pd.read_csv(\"../project_catalog (1)-Copy1.csv\")\n",
    "hmp_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2915, 17)"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmp_df.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "azdata_cell_guid": "e172475a-c4ee-414a-8367-9965355dbba6"
   },
   "source": [
    "## Missing Data\n",
    "\n",
    "Test your dataset for missing data and handle it as needed. Make notes in the form of code comments as to your thought process."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "azdata_cell_guid": "e1dc66ef-e471-4c27-92e7-ee878c106eba"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HMP ID</th>\n",
       "      <th>GOLD ID</th>\n",
       "      <th>Organism Name</th>\n",
       "      <th>Domain</th>\n",
       "      <th>NCBI Superkingdom</th>\n",
       "      <th>HMP Isolation Body Site</th>\n",
       "      <th>Project Status</th>\n",
       "      <th>Current Finishing Level</th>\n",
       "      <th>NCBI Submission Status</th>\n",
       "      <th>NCBI Project ID</th>\n",
       "      <th>Genbank ID</th>\n",
       "      <th>Gene Count</th>\n",
       "      <th>IMG/HMP ID</th>\n",
       "      <th>HOMD ID</th>\n",
       "      <th>Sequencing Center</th>\n",
       "      <th>Funding Source</th>\n",
       "      <th>Strain Repository ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2910</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2911</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2912</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2913</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2914</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2915 rows Ã— 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      HMP ID  GOLD ID  Organism Name  Domain  NCBI Superkingdom  \\\n",
       "0      False    False          False   False              False   \n",
       "1      False    False          False   False              False   \n",
       "2      False    False          False   False              False   \n",
       "3      False    False          False   False              False   \n",
       "4      False    False          False   False              False   \n",
       "...      ...      ...            ...     ...                ...   \n",
       "2910   False    False          False   False              False   \n",
       "2911   False    False          False   False              False   \n",
       "2912   False    False          False   False              False   \n",
       "2913   False    False          False   False              False   \n",
       "2914   False    False          False   False              False   \n",
       "\n",
       "      HMP Isolation Body Site  Project Status  Current Finishing Level  \\\n",
       "0                       False           False                    False   \n",
       "1                       False           False                    False   \n",
       "2                       False           False                    False   \n",
       "3                       False           False                    False   \n",
       "4                       False           False                    False   \n",
       "...                       ...             ...                      ...   \n",
       "2910                    False           False                    False   \n",
       "2911                    False           False                    False   \n",
       "2912                    False           False                    False   \n",
       "2913                    False           False                    False   \n",
       "2914                    False           False                    False   \n",
       "\n",
       "      NCBI Submission Status  NCBI Project ID  Genbank ID  Gene Count  \\\n",
       "0                      False            False       False       False   \n",
       "1                      False            False       False       False   \n",
       "2                      False            False       False       False   \n",
       "3                      False            False       False       False   \n",
       "4                      False            False       False       False   \n",
       "...                      ...              ...         ...         ...   \n",
       "2910                   False            False       False       False   \n",
       "2911                   False            False       False       False   \n",
       "2912                   False            False       False       False   \n",
       "2913                   False            False       False       False   \n",
       "2914                   False            False       False       False   \n",
       "\n",
       "      IMG/HMP ID  HOMD ID  Sequencing Center  Funding Source  \\\n",
       "0          False    False              False           False   \n",
       "1          False     True              False           False   \n",
       "2          False    False              False           False   \n",
       "3          False    False              False           False   \n",
       "4          False     True              False           False   \n",
       "...          ...      ...                ...             ...   \n",
       "2910       False     True              False           False   \n",
       "2911       False     True              False           False   \n",
       "2912       False     True              False           False   \n",
       "2913       False     True              False           False   \n",
       "2914       False     True              False           False   \n",
       "\n",
       "      Strain Repository ID  \n",
       "0                    False  \n",
       "1                    False  \n",
       "2                    False  \n",
       "3                    False  \n",
       "4                    False  \n",
       "...                    ...  \n",
       "2910                 False  \n",
       "2911                 False  \n",
       "2912                 False  \n",
       "2913                 False  \n",
       "2914                 False  \n",
       "\n",
       "[2915 rows x 17 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmp_df.isnull()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The results of the .isnull() check reveal only null values present in the HOMD ID column. Looking into the datasource lets me know that this column is an ID from [the human oral microbiome database](http://www.homd.org/index.php?name=HOMD&view=dynamic&oraltaxonid=389).\n",
    "I don't plan on using this ID to tackle the business issue, so I'll delete the column altogether."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 2915 entries, 0 to 2914\n",
      "Data columns (total 16 columns):\n",
      " #   Column                   Non-Null Count  Dtype \n",
      "---  ------                   --------------  ----- \n",
      " 0   HMP ID                   2915 non-null   int64 \n",
      " 1   GOLD ID                  1783 non-null   object\n",
      " 2   Organism Name            2915 non-null   object\n",
      " 3   Domain                   2712 non-null   object\n",
      " 4   NCBI Superkingdom        2751 non-null   object\n",
      " 5   HMP Isolation Body Site  2915 non-null   object\n",
      " 6   Project Status           2915 non-null   object\n",
      " 7   Current Finishing Level  1579 non-null   object\n",
      " 8   NCBI Submission Status   2915 non-null   object\n",
      " 9   NCBI Project ID          2915 non-null   int64 \n",
      " 10  Genbank ID               1579 non-null   object\n",
      " 11  Gene Count               2915 non-null   int64 \n",
      " 12  IMG/HMP ID               2915 non-null   int64 \n",
      " 13  Sequencing Center        2911 non-null   object\n",
      " 14  Funding Source           2915 non-null   object\n",
      " 15  Strain Repository ID     1377 non-null   object\n",
      "dtypes: int64(4), object(12)\n",
      "memory usage: 364.5+ KB\n"
     ]
    }
   ],
   "source": [
    "hmp_df = hmp_df.drop('HOMD ID', axis = 1)\n",
    "hmp_df.info()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The output above shows me that the HOMD ID column was dropped. The output only contains a summary of the rows, however. I should find the sum of the null values present in the whole dataframe to be sure that there are not other null values in other columns not showing up in the summary above."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "HMP ID                        0\n",
       "GOLD ID                    1132\n",
       "Organism Name                 0\n",
       "Domain                      203\n",
       "NCBI Superkingdom           164\n",
       "HMP Isolation Body Site       0\n",
       "Project Status                0\n",
       "Current Finishing Level    1336\n",
       "NCBI Submission Status        0\n",
       "NCBI Project ID               0\n",
       "Genbank ID                 1336\n",
       "Gene Count                    0\n",
       "IMG/HMP ID                    0\n",
       "Sequencing Center             4\n",
       "Funding Source                0\n",
       "Strain Repository ID       1538\n",
       "dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmp_df.isnull().sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Indeed there are several other columns containing null values. These are: GOLD ID, Domain, NCBI Superkingdom, Current Finishing level, Genbank ID, and Strain Repository ID.\n",
    "From this list, the only columns I remain interested in exploring are Domain and NCBI Superkingdom,. The rest I do not need in my analysis. I'll drop them."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "hmp_df = hmp_df.drop(['GOLD ID', 'Current Finishing Level', 'Genbank ID', 'Strain Repository ID'], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 2915 entries, 0 to 2914\n",
      "Data columns (total 12 columns):\n",
      " #   Column                   Non-Null Count  Dtype \n",
      "---  ------                   --------------  ----- \n",
      " 0   HMP ID                   2915 non-null   int64 \n",
      " 1   Organism Name            2915 non-null   object\n",
      " 2   Domain                   2712 non-null   object\n",
      " 3   NCBI Superkingdom        2751 non-null   object\n",
      " 4   HMP Isolation Body Site  2915 non-null   object\n",
      " 5   Project Status           2915 non-null   object\n",
      " 6   NCBI Submission Status   2915 non-null   object\n",
      " 7   NCBI Project ID          2915 non-null   int64 \n",
      " 8   Gene Count               2915 non-null   int64 \n",
      " 9   IMG/HMP ID               2915 non-null   int64 \n",
      " 10  Sequencing Center        2911 non-null   object\n",
      " 11  Funding Source           2915 non-null   object\n",
      "dtypes: int64(4), object(8)\n",
      "memory usage: 273.4+ KB\n"
     ]
    }
   ],
   "source": [
    "hmp_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "HMP ID                       0\n",
       "Organism Name                0\n",
       "Domain                     203\n",
       "NCBI Superkingdom          164\n",
       "HMP Isolation Body Site      0\n",
       "Project Status               0\n",
       "NCBI Submission Status       0\n",
       "NCBI Project ID              0\n",
       "Gene Count                   0\n",
       "IMG/HMP ID                   0\n",
       "Sequencing Center            4\n",
       "Funding Source               0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmp_df.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "HMP ID                     0\n",
       "Organism Name              0\n",
       "Domain                     0\n",
       "NCBI Superkingdom          0\n",
       "HMP Isolation Body Site    0\n",
       "Project Status             0\n",
       "NCBI Submission Status     0\n",
       "NCBI Project ID            0\n",
       "Gene Count                 0\n",
       "IMG/HMP ID                 0\n",
       "Sequencing Center          4\n",
       "Funding Source             0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmp_df['Domain']=hmp_df['Domain'].fillna(value='NO DOMAIN')\n",
    "hmp_df['NCBI Superkingdom']=hmp_df['NCBI Superkingdom'].fillna(value='NO KINGDOM')\n",
    "hmp_df.isnull().sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After filling in the null domain and superkingdom elements, I see that i missed another column with a few null values: 'Sequencing Center'. I dont expect to need this information for the project either so I'll remove it."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 2915 entries, 0 to 2914\n",
      "Data columns (total 11 columns):\n",
      " #   Column                   Non-Null Count  Dtype \n",
      "---  ------                   --------------  ----- \n",
      " 0   HMP ID                   2915 non-null   int64 \n",
      " 1   Organism Name            2915 non-null   object\n",
      " 2   Domain                   2915 non-null   object\n",
      " 3   NCBI Superkingdom        2915 non-null   object\n",
      " 4   HMP Isolation Body Site  2915 non-null   object\n",
      " 5   Project Status           2915 non-null   object\n",
      " 6   NCBI Submission Status   2915 non-null   object\n",
      " 7   NCBI Project ID          2915 non-null   int64 \n",
      " 8   Gene Count               2915 non-null   int64 \n",
      " 9   IMG/HMP ID               2915 non-null   int64 \n",
      " 10  Funding Source           2915 non-null   object\n",
      "dtypes: int64(4), object(7)\n",
      "memory usage: 250.6+ KB\n"
     ]
    }
   ],
   "source": [
    "hmp_df=hmp_df.drop('Sequencing Center', axis=1)\n",
    "hmp_df.info()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "azdata_cell_guid": "1233f543-e9a0-4f78-96f5-d7536554102e"
   },
   "source": [
    "## Irregular Data\n",
    "\n",
    "Detect outliers in your dataset and handle them as needed. Use code comments to make notes about your thought process."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "azdata_cell_guid": "efed50ae-16f0-471d-98e2-632553a74c12"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HMP ID</th>\n",
       "      <th>NCBI Project ID</th>\n",
       "      <th>Gene Count</th>\n",
       "      <th>IMG/HMP ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>2915.000000</td>\n",
       "      <td>2915.000000</td>\n",
       "      <td>2915.000000</td>\n",
       "      <td>2.915000e+03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>3193.047684</td>\n",
       "      <td>152890.012007</td>\n",
       "      <td>1474.622642</td>\n",
       "      <td>2.250715e+08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>3055.524395</td>\n",
       "      <td>104857.664414</td>\n",
       "      <td>1653.762393</td>\n",
       "      <td>5.295180e+08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>1262.500000</td>\n",
       "      <td>49365.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>2122.000000</td>\n",
       "      <td>169465.000000</td>\n",
       "      <td>1517.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>3096.500000</td>\n",
       "      <td>269820.500000</td>\n",
       "      <td>2489.500000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>9999.000000</td>\n",
       "      <td>311149.000000</td>\n",
       "      <td>8490.000000</td>\n",
       "      <td>2.524023e+09</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            HMP ID  NCBI Project ID   Gene Count    IMG/HMP ID\n",
       "count  2915.000000      2915.000000  2915.000000  2.915000e+03\n",
       "mean   3193.047684    152890.012007  1474.622642  2.250715e+08\n",
       "std    3055.524395    104857.664414  1653.762393  5.295180e+08\n",
       "min       1.000000         0.000000     0.000000  0.000000e+00\n",
       "25%    1262.500000     49365.000000     0.000000  0.000000e+00\n",
       "50%    2122.000000    169465.000000  1517.000000  0.000000e+00\n",
       "75%    3096.500000    269820.500000  2489.500000  0.000000e+00\n",
       "max    9999.000000    311149.000000  8490.000000  2.524023e+09"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmp_df.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The only numerical column that would have outliers is Gene Count. The others are IDs. I know from my EDA workbook (checkpoint 2) what the distribution of Gene Count looks like. I dont not wish to remove any outliers at this time."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "azdata_cell_guid": "6f5b8ee0-bab3-44bc-958a-67d1e4c0407f"
   },
   "source": [
    "## Unnecessary Data\n",
    "\n",
    "Look for the different types of unnecessary data in your dataset and address it as needed. Make sure to use code comments to illustrate your thought process."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "azdata_cell_guid": "e788a239-2fbf-41de-9bd3-19e52e3b187c"
   },
   "source": [
    "I've removed several columns that contained null values that I know I am not interested in exploring further from the perspective of my business issue. Most of these were alternative identifier figures from other databases for the organisms studied. I probably will also not be interested in doing any analysis with submission status, project project status, or funding source. However, I'll keep them for now in case they prove to be interesting figures to work with in Tableau."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "azdata_cell_guid": "53e0cf94-c68a-4fa0-9849-9505a66bcce6"
   },
   "source": [
    "## Inconsistent Data\n",
    "\n",
    "Check for inconsistent data and address any that arises. As always, use code comments to illustrate your thought process."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "azdata_cell_guid": "e9de6624-812a-43f8-8e20-93b4a49b091f"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HMP ID</th>\n",
       "      <th>Organism Name</th>\n",
       "      <th>Domain</th>\n",
       "      <th>NCBI Superkingdom</th>\n",
       "      <th>HMP Isolation Body Site</th>\n",
       "      <th>Project Status</th>\n",
       "      <th>NCBI Submission Status</th>\n",
       "      <th>NCBI Project ID</th>\n",
       "      <th>Gene Count</th>\n",
       "      <th>IMG/HMP ID</th>\n",
       "      <th>Funding Source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Abiotrophia defectiva ATCC 49176</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>oral</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>33011</td>\n",
       "      <td>1950</td>\n",
       "      <td>643886181</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>Achromobacter piechaudii ATCC 43553</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>airways</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>46343</td>\n",
       "      <td>5755</td>\n",
       "      <td>647000200</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>Achromobacter xylosoxidans C54</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>airways</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38739</td>\n",
       "      <td>6010</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>Acinetobacter baumannii ATCC 19606</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>urogenital_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38509</td>\n",
       "      <td>3832</td>\n",
       "      <td>647533101</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12</td>\n",
       "      <td>Acinetobacter calcoaceticus RUH2202</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>skin</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38337</td>\n",
       "      <td>3632</td>\n",
       "      <td>646206267</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>13</td>\n",
       "      <td>Acinetobacter sp. SH024</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>skin</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>41947</td>\n",
       "      <td>3703</td>\n",
       "      <td>647533108</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>14</td>\n",
       "      <td>Acinetobacter sp. RUH2624</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>skin</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38511</td>\n",
       "      <td>3693</td>\n",
       "      <td>647533107</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>15</td>\n",
       "      <td>Acinetobacter haemolyticus ATCC 19194</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>airways</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>43187</td>\n",
       "      <td>3491</td>\n",
       "      <td>647000202</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>16</td>\n",
       "      <td>Acinetobacter johnsonii SH046</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>skin</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38339</td>\n",
       "      <td>3434</td>\n",
       "      <td>646206268</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>17</td>\n",
       "      <td>Acinetobacter lwoffii SH145</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>skin</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38343</td>\n",
       "      <td>3202</td>\n",
       "      <td>646206270</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>18</td>\n",
       "      <td>Acinetobacter radioresistens SH164</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38345</td>\n",
       "      <td>2938</td>\n",
       "      <td>646206271</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>19</td>\n",
       "      <td>Acinetobacter radioresistens SK82</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>skin</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>34081</td>\n",
       "      <td>3188</td>\n",
       "      <td>645058758</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>20</td>\n",
       "      <td>Acinetobacter baumannii 6013113</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>skin</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>33017</td>\n",
       "      <td>4010</td>\n",
       "      <td>647000203</td>\n",
       "      <td>NIH-NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>21</td>\n",
       "      <td>Acinetobacter baumannii 6013150</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>skin</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>33071</td>\n",
       "      <td>3964</td>\n",
       "      <td>647000204</td>\n",
       "      <td>NIH-NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>22</td>\n",
       "      <td>Acinetobacter baumannii 6014059</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>skin</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>33073</td>\n",
       "      <td>3880</td>\n",
       "      <td>647000205</td>\n",
       "      <td>NIH-NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>23</td>\n",
       "      <td>Acinetobacter sp. ATCC 27244</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>skin</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>30949</td>\n",
       "      <td>3327</td>\n",
       "      <td>643886214</td>\n",
       "      <td>NIH-NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>26</td>\n",
       "      <td>Acinetobacter junii SH205</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38341</td>\n",
       "      <td>3252</td>\n",
       "      <td>646206269</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>27</td>\n",
       "      <td>Actinobacillus ureae ATCC 25976</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>airways</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>52977</td>\n",
       "      <td>2475</td>\n",
       "      <td>649989900</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>43</td>\n",
       "      <td>Actinobaculum P1 oral taxon 183 F0552</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>NO KINGDOM</td>\n",
       "      <td>oral</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>173932</td>\n",
       "      <td>2320</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH-HMP Sequencing Center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>44</td>\n",
       "      <td>Actinomyces coleocanis DSM 15436</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>urogenital_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>31427</td>\n",
       "      <td>1546</td>\n",
       "      <td>643886017</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>45</td>\n",
       "      <td>Actinomyces graevenitzii C83</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>airways</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38761</td>\n",
       "      <td>1853</td>\n",
       "      <td>2513237375</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>50</td>\n",
       "      <td>Actinomyces odontolyticus ATCC 17982</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>oral</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>18171</td>\n",
       "      <td>2159</td>\n",
       "      <td>640963058</td>\n",
       "      <td>NIH-HMP NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>51</td>\n",
       "      <td>Actinomyces odontolyticus C702</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>airways</td>\n",
       "      <td>In Progress</td>\n",
       "      <td>4. sequence public on NCBI site</td>\n",
       "      <td>38741</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>58</td>\n",
       "      <td>Actinomyces urogenitalis DSM 15434</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>urogenital_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>31429</td>\n",
       "      <td>2403</td>\n",
       "      <td>643886015</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>59</td>\n",
       "      <td>Actinomyces viscosus C505</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>airways</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38743</td>\n",
       "      <td>2587</td>\n",
       "      <td>649989905</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>61</td>\n",
       "      <td>Aerococcus viridans ATCC 11563</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>urogenital_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38729</td>\n",
       "      <td>1929</td>\n",
       "      <td>647000207</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>63</td>\n",
       "      <td>Aeromicrobium marinum DSM 15272</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>oral</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>34545</td>\n",
       "      <td>3081</td>\n",
       "      <td>645951844</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>67</td>\n",
       "      <td>Anaerovorax sp. ORNL_6EZ5-Gt_3_Pl1-27</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>unknown</td>\n",
       "      <td>In Progress</td>\n",
       "      <td>2. project has NCBI project id</td>\n",
       "      <td>217536</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH-HMP Sequencing Center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>68</td>\n",
       "      <td>Alistipes sp. 10_1_37FAA</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>In Progress</td>\n",
       "      <td>4. sequence public on NCBI site</td>\n",
       "      <td>41949</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH-NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>69</td>\n",
       "      <td>Alistipes putredinis DSM 17216</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>19655</td>\n",
       "      <td>2742</td>\n",
       "      <td>641736205</td>\n",
       "      <td>NIH-HMP NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>70</td>\n",
       "      <td>Anaerovorax sp. ORNL_6EZ5-Gt_3_Pl2-127</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>unknown</td>\n",
       "      <td>In Progress</td>\n",
       "      <td>2. project has NCBI project id</td>\n",
       "      <td>217537</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH-HMP Sequencing Center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>71</td>\n",
       "      <td>Anaerococcus hydrogenalis DSM 7454</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>30747</td>\n",
       "      <td>2069</td>\n",
       "      <td>642979323</td>\n",
       "      <td>NIH-HMP NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>72</td>\n",
       "      <td>Anaerococcus lactolyticus ATCC 51172</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>urogenital_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>30951</td>\n",
       "      <td>2253</td>\n",
       "      <td>643886145</td>\n",
       "      <td>NIH-NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>76</td>\n",
       "      <td>Anaerococcus sp. S9 PR-5</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>urogenital_tract</td>\n",
       "      <td>In Progress</td>\n",
       "      <td>4. sequence public on NCBI site</td>\n",
       "      <td>219648</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH-HMP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>77</td>\n",
       "      <td>Anaerococcus tetradius ATCC 35098</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>urogenital_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>31433</td>\n",
       "      <td>2079</td>\n",
       "      <td>643886055</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>78</td>\n",
       "      <td>Anaerococcus vaginalis ATCC 51170</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>skin</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38727</td>\n",
       "      <td>1764</td>\n",
       "      <td>645058761</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>79</td>\n",
       "      <td>Anaerofustis stercorihominis DSM 17244</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>19657</td>\n",
       "      <td>2331</td>\n",
       "      <td>641736193</td>\n",
       "      <td>NIH-HMP NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>80</td>\n",
       "      <td>Anaeroglobus geminatus F0357</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>oral</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>50397</td>\n",
       "      <td>2148</td>\n",
       "      <td>2513237300</td>\n",
       "      <td>NIH-HMP Sequencing Center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>81</td>\n",
       "      <td>Anaerostipes caccae DSM 14662</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>18213</td>\n",
       "      <td>3748</td>\n",
       "      <td>641736227</td>\n",
       "      <td>NIH-HMP NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>82</td>\n",
       "      <td>Anaerotruncus colihominis DSM 17241</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>19659</td>\n",
       "      <td>4427</td>\n",
       "      <td>641736271</td>\n",
       "      <td>NIH-HMP NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>83</td>\n",
       "      <td>Aneurinibacillus aneurinilyticus ATCC 12856</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38313</td>\n",
       "      <td>6214</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>86</td>\n",
       "      <td>Atopobium rimae ATCC 49626</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>oral</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>30999</td>\n",
       "      <td>1548</td>\n",
       "      <td>643886019</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>91</td>\n",
       "      <td>Atopobium vaginae DSM 15829</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>urogenital_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>31435</td>\n",
       "      <td>1244</td>\n",
       "      <td>643886063</td>\n",
       "      <td>NIH-HMP Jumpstart Supplement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>101</td>\n",
       "      <td>Bacteroides sp. 2_1_16</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38347</td>\n",
       "      <td>4683</td>\n",
       "      <td>647533110</td>\n",
       "      <td>NIH-NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>102</td>\n",
       "      <td>Bacteroides sp. 2_1_22</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>38349</td>\n",
       "      <td>4814</td>\n",
       "      <td>647533111</td>\n",
       "      <td>NIH-NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>104</td>\n",
       "      <td>Bacteroides sp. 3_1_19</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>41951</td>\n",
       "      <td>4328</td>\n",
       "      <td>648861002</td>\n",
       "      <td>NIH-NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>107</td>\n",
       "      <td>Bacteroidales bacterium ORNL_2OR_SD_356</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>unknown</td>\n",
       "      <td>In Progress</td>\n",
       "      <td>2. project has NCBI project id</td>\n",
       "      <td>217538</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH-HMP Sequencing Center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>108</td>\n",
       "      <td>Bacteroidetes bacterium ORNL_1UT_ORNL_28</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>unknown</td>\n",
       "      <td>In Progress</td>\n",
       "      <td>2. project has NCBI project id</td>\n",
       "      <td>217539</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NIH-HMP Sequencing Center</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>109</td>\n",
       "      <td>Bacteroides caccae ATCC 43185</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>18163</td>\n",
       "      <td>3855</td>\n",
       "      <td>640963023</td>\n",
       "      <td>NIH-HMP NHGRI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>110</td>\n",
       "      <td>Bacteroides capillosus ATCC 29799</td>\n",
       "      <td>BACTERIAL</td>\n",
       "      <td>Bacteria</td>\n",
       "      <td>gastrointestinal_tract</td>\n",
       "      <td>Complete</td>\n",
       "      <td>6. annotation (and sequence) public on NCBI site</td>\n",
       "      <td>18173</td>\n",
       "      <td>4833</td>\n",
       "      <td>640963014</td>\n",
       "      <td>NIH-HMP NHGRI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    HMP ID                                Organism Name     Domain  \\\n",
       "0        1             Abiotrophia defectiva ATCC 49176  BACTERIAL   \n",
       "1        4          Achromobacter piechaudii ATCC 43553  BACTERIAL   \n",
       "2        5               Achromobacter xylosoxidans C54  BACTERIAL   \n",
       "3       10           Acinetobacter baumannii ATCC 19606  BACTERIAL   \n",
       "4       12          Acinetobacter calcoaceticus RUH2202  BACTERIAL   \n",
       "5       13                      Acinetobacter sp. SH024  BACTERIAL   \n",
       "6       14                    Acinetobacter sp. RUH2624  BACTERIAL   \n",
       "7       15        Acinetobacter haemolyticus ATCC 19194  BACTERIAL   \n",
       "8       16                Acinetobacter johnsonii SH046  BACTERIAL   \n",
       "9       17                  Acinetobacter lwoffii SH145  BACTERIAL   \n",
       "10      18           Acinetobacter radioresistens SH164  BACTERIAL   \n",
       "11      19            Acinetobacter radioresistens SK82  BACTERIAL   \n",
       "12      20              Acinetobacter baumannii 6013113  BACTERIAL   \n",
       "13      21              Acinetobacter baumannii 6013150  BACTERIAL   \n",
       "14      22              Acinetobacter baumannii 6014059  BACTERIAL   \n",
       "15      23                 Acinetobacter sp. ATCC 27244  BACTERIAL   \n",
       "16      26                    Acinetobacter junii SH205  BACTERIAL   \n",
       "17      27              Actinobacillus ureae ATCC 25976  BACTERIAL   \n",
       "18      43        Actinobaculum P1 oral taxon 183 F0552  BACTERIAL   \n",
       "19      44             Actinomyces coleocanis DSM 15436  BACTERIAL   \n",
       "20      45                 Actinomyces graevenitzii C83  BACTERIAL   \n",
       "21      50         Actinomyces odontolyticus ATCC 17982  BACTERIAL   \n",
       "22      51               Actinomyces odontolyticus C702  BACTERIAL   \n",
       "23      58           Actinomyces urogenitalis DSM 15434  BACTERIAL   \n",
       "24      59                    Actinomyces viscosus C505  BACTERIAL   \n",
       "25      61               Aerococcus viridans ATCC 11563  BACTERIAL   \n",
       "26      63              Aeromicrobium marinum DSM 15272  BACTERIAL   \n",
       "27      67        Anaerovorax sp. ORNL_6EZ5-Gt_3_Pl1-27  BACTERIAL   \n",
       "28      68                     Alistipes sp. 10_1_37FAA  BACTERIAL   \n",
       "29      69               Alistipes putredinis DSM 17216  BACTERIAL   \n",
       "30      70       Anaerovorax sp. ORNL_6EZ5-Gt_3_Pl2-127  BACTERIAL   \n",
       "31      71           Anaerococcus hydrogenalis DSM 7454  BACTERIAL   \n",
       "32      72         Anaerococcus lactolyticus ATCC 51172  BACTERIAL   \n",
       "33      76                     Anaerococcus sp. S9 PR-5  BACTERIAL   \n",
       "34      77            Anaerococcus tetradius ATCC 35098  BACTERIAL   \n",
       "35      78            Anaerococcus vaginalis ATCC 51170  BACTERIAL   \n",
       "36      79       Anaerofustis stercorihominis DSM 17244  BACTERIAL   \n",
       "37      80                 Anaeroglobus geminatus F0357  BACTERIAL   \n",
       "38      81                Anaerostipes caccae DSM 14662  BACTERIAL   \n",
       "39      82          Anaerotruncus colihominis DSM 17241  BACTERIAL   \n",
       "40      83  Aneurinibacillus aneurinilyticus ATCC 12856  BACTERIAL   \n",
       "41      86                   Atopobium rimae ATCC 49626  BACTERIAL   \n",
       "42      91                  Atopobium vaginae DSM 15829  BACTERIAL   \n",
       "43     101                       Bacteroides sp. 2_1_16  BACTERIAL   \n",
       "44     102                       Bacteroides sp. 2_1_22  BACTERIAL   \n",
       "45     104                       Bacteroides sp. 3_1_19  BACTERIAL   \n",
       "46     107      Bacteroidales bacterium ORNL_2OR_SD_356  BACTERIAL   \n",
       "47     108     Bacteroidetes bacterium ORNL_1UT_ORNL_28  BACTERIAL   \n",
       "48     109                Bacteroides caccae ATCC 43185  BACTERIAL   \n",
       "49     110            Bacteroides capillosus ATCC 29799  BACTERIAL   \n",
       "\n",
       "   NCBI Superkingdom HMP Isolation Body Site Project Status  \\\n",
       "0           Bacteria                    oral       Complete   \n",
       "1           Bacteria                 airways       Complete   \n",
       "2           Bacteria                 airways       Complete   \n",
       "3           Bacteria        urogenital_tract       Complete   \n",
       "4           Bacteria                    skin       Complete   \n",
       "5           Bacteria                    skin       Complete   \n",
       "6           Bacteria                    skin       Complete   \n",
       "7           Bacteria                 airways       Complete   \n",
       "8           Bacteria                    skin       Complete   \n",
       "9           Bacteria                    skin       Complete   \n",
       "10          Bacteria  gastrointestinal_tract       Complete   \n",
       "11          Bacteria                    skin       Complete   \n",
       "12          Bacteria                    skin       Complete   \n",
       "13          Bacteria                    skin       Complete   \n",
       "14          Bacteria                    skin       Complete   \n",
       "15          Bacteria                    skin       Complete   \n",
       "16          Bacteria  gastrointestinal_tract       Complete   \n",
       "17          Bacteria                 airways       Complete   \n",
       "18        NO KINGDOM                    oral       Complete   \n",
       "19          Bacteria        urogenital_tract       Complete   \n",
       "20          Bacteria                 airways       Complete   \n",
       "21          Bacteria                    oral       Complete   \n",
       "22          Bacteria                 airways    In Progress   \n",
       "23          Bacteria        urogenital_tract       Complete   \n",
       "24          Bacteria                 airways       Complete   \n",
       "25          Bacteria        urogenital_tract       Complete   \n",
       "26          Bacteria                    oral       Complete   \n",
       "27          Bacteria                 unknown    In Progress   \n",
       "28          Bacteria  gastrointestinal_tract    In Progress   \n",
       "29          Bacteria  gastrointestinal_tract       Complete   \n",
       "30          Bacteria                 unknown    In Progress   \n",
       "31          Bacteria  gastrointestinal_tract       Complete   \n",
       "32          Bacteria        urogenital_tract       Complete   \n",
       "33          Bacteria        urogenital_tract    In Progress   \n",
       "34          Bacteria        urogenital_tract       Complete   \n",
       "35          Bacteria                    skin       Complete   \n",
       "36          Bacteria  gastrointestinal_tract       Complete   \n",
       "37          Bacteria                    oral       Complete   \n",
       "38          Bacteria  gastrointestinal_tract       Complete   \n",
       "39          Bacteria  gastrointestinal_tract       Complete   \n",
       "40          Bacteria  gastrointestinal_tract       Complete   \n",
       "41          Bacteria                    oral       Complete   \n",
       "42          Bacteria        urogenital_tract       Complete   \n",
       "43          Bacteria  gastrointestinal_tract       Complete   \n",
       "44          Bacteria  gastrointestinal_tract       Complete   \n",
       "45          Bacteria  gastrointestinal_tract       Complete   \n",
       "46          Bacteria                 unknown    In Progress   \n",
       "47          Bacteria                 unknown    In Progress   \n",
       "48          Bacteria  gastrointestinal_tract       Complete   \n",
       "49          Bacteria  gastrointestinal_tract       Complete   \n",
       "\n",
       "                              NCBI Submission Status  NCBI Project ID  \\\n",
       "0   6. annotation (and sequence) public on NCBI site            33011   \n",
       "1   6. annotation (and sequence) public on NCBI site            46343   \n",
       "2   6. annotation (and sequence) public on NCBI site            38739   \n",
       "3   6. annotation (and sequence) public on NCBI site            38509   \n",
       "4   6. annotation (and sequence) public on NCBI site            38337   \n",
       "5   6. annotation (and sequence) public on NCBI site            41947   \n",
       "6   6. annotation (and sequence) public on NCBI site            38511   \n",
       "7   6. annotation (and sequence) public on NCBI site            43187   \n",
       "8   6. annotation (and sequence) public on NCBI site            38339   \n",
       "9   6. annotation (and sequence) public on NCBI site            38343   \n",
       "10  6. annotation (and sequence) public on NCBI site            38345   \n",
       "11  6. annotation (and sequence) public on NCBI site            34081   \n",
       "12  6. annotation (and sequence) public on NCBI site            33017   \n",
       "13  6. annotation (and sequence) public on NCBI site            33071   \n",
       "14  6. annotation (and sequence) public on NCBI site            33073   \n",
       "15  6. annotation (and sequence) public on NCBI site            30949   \n",
       "16  6. annotation (and sequence) public on NCBI site            38341   \n",
       "17  6. annotation (and sequence) public on NCBI site            52977   \n",
       "18  6. annotation (and sequence) public on NCBI site           173932   \n",
       "19  6. annotation (and sequence) public on NCBI site            31427   \n",
       "20  6. annotation (and sequence) public on NCBI site            38761   \n",
       "21  6. annotation (and sequence) public on NCBI site            18171   \n",
       "22                   4. sequence public on NCBI site            38741   \n",
       "23  6. annotation (and sequence) public on NCBI site            31429   \n",
       "24  6. annotation (and sequence) public on NCBI site            38743   \n",
       "25  6. annotation (and sequence) public on NCBI site            38729   \n",
       "26  6. annotation (and sequence) public on NCBI site            34545   \n",
       "27                    2. project has NCBI project id           217536   \n",
       "28                   4. sequence public on NCBI site            41949   \n",
       "29  6. annotation (and sequence) public on NCBI site            19655   \n",
       "30                    2. project has NCBI project id           217537   \n",
       "31  6. annotation (and sequence) public on NCBI site            30747   \n",
       "32  6. annotation (and sequence) public on NCBI site            30951   \n",
       "33                   4. sequence public on NCBI site           219648   \n",
       "34  6. annotation (and sequence) public on NCBI site            31433   \n",
       "35  6. annotation (and sequence) public on NCBI site            38727   \n",
       "36  6. annotation (and sequence) public on NCBI site            19657   \n",
       "37  6. annotation (and sequence) public on NCBI site            50397   \n",
       "38  6. annotation (and sequence) public on NCBI site            18213   \n",
       "39  6. annotation (and sequence) public on NCBI site            19659   \n",
       "40  6. annotation (and sequence) public on NCBI site            38313   \n",
       "41  6. annotation (and sequence) public on NCBI site            30999   \n",
       "42  6. annotation (and sequence) public on NCBI site            31435   \n",
       "43  6. annotation (and sequence) public on NCBI site            38347   \n",
       "44  6. annotation (and sequence) public on NCBI site            38349   \n",
       "45  6. annotation (and sequence) public on NCBI site            41951   \n",
       "46                    2. project has NCBI project id           217538   \n",
       "47                    2. project has NCBI project id           217539   \n",
       "48  6. annotation (and sequence) public on NCBI site            18163   \n",
       "49  6. annotation (and sequence) public on NCBI site            18173   \n",
       "\n",
       "    Gene Count  IMG/HMP ID                Funding Source  \n",
       "0         1950   643886181  NIH-HMP Jumpstart Supplement  \n",
       "1         5755   647000200  NIH-HMP Jumpstart Supplement  \n",
       "2         6010           0  NIH-HMP Jumpstart Supplement  \n",
       "3         3832   647533101  NIH-HMP Jumpstart Supplement  \n",
       "4         3632   646206267  NIH-HMP Jumpstart Supplement  \n",
       "5         3703   647533108  NIH-HMP Jumpstart Supplement  \n",
       "6         3693   647533107  NIH-HMP Jumpstart Supplement  \n",
       "7         3491   647000202  NIH-HMP Jumpstart Supplement  \n",
       "8         3434   646206268  NIH-HMP Jumpstart Supplement  \n",
       "9         3202   646206270  NIH-HMP Jumpstart Supplement  \n",
       "10        2938   646206271  NIH-HMP Jumpstart Supplement  \n",
       "11        3188   645058758  NIH-HMP Jumpstart Supplement  \n",
       "12        4010   647000203                     NIH-NHGRI  \n",
       "13        3964   647000204                     NIH-NHGRI  \n",
       "14        3880   647000205                     NIH-NHGRI  \n",
       "15        3327   643886214                     NIH-NHGRI  \n",
       "16        3252   646206269  NIH-HMP Jumpstart Supplement  \n",
       "17        2475   649989900  NIH-HMP Jumpstart Supplement  \n",
       "18        2320           0     NIH-HMP Sequencing Center  \n",
       "19        1546   643886017  NIH-HMP Jumpstart Supplement  \n",
       "20        1853  2513237375  NIH-HMP Jumpstart Supplement  \n",
       "21        2159   640963058                 NIH-HMP NHGRI  \n",
       "22           0           0  NIH-HMP Jumpstart Supplement  \n",
       "23        2403   643886015  NIH-HMP Jumpstart Supplement  \n",
       "24        2587   649989905  NIH-HMP Jumpstart Supplement  \n",
       "25        1929   647000207  NIH-HMP Jumpstart Supplement  \n",
       "26        3081   645951844  NIH-HMP Jumpstart Supplement  \n",
       "27           0           0     NIH-HMP Sequencing Center  \n",
       "28           0           0                     NIH-NHGRI  \n",
       "29        2742   641736205                 NIH-HMP NHGRI  \n",
       "30           0           0     NIH-HMP Sequencing Center  \n",
       "31        2069   642979323                 NIH-HMP NHGRI  \n",
       "32        2253   643886145                     NIH-NHGRI  \n",
       "33           0           0                       NIH-HMP  \n",
       "34        2079   643886055  NIH-HMP Jumpstart Supplement  \n",
       "35        1764   645058761  NIH-HMP Jumpstart Supplement  \n",
       "36        2331   641736193                 NIH-HMP NHGRI  \n",
       "37        2148  2513237300     NIH-HMP Sequencing Center  \n",
       "38        3748   641736227                 NIH-HMP NHGRI  \n",
       "39        4427   641736271                 NIH-HMP NHGRI  \n",
       "40        6214           0  NIH-HMP Jumpstart Supplement  \n",
       "41        1548   643886019  NIH-HMP Jumpstart Supplement  \n",
       "42        1244   643886063  NIH-HMP Jumpstart Supplement  \n",
       "43        4683   647533110                     NIH-NHGRI  \n",
       "44        4814   647533111                     NIH-NHGRI  \n",
       "45        4328   648861002                     NIH-NHGRI  \n",
       "46           0           0     NIH-HMP Sequencing Center  \n",
       "47           0           0     NIH-HMP Sequencing Center  \n",
       "48        3855   640963023                 NIH-HMP NHGRI  \n",
       "49        4833   640963014                 NIH-HMP NHGRI  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmp_df.head(50)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From a look at the first 50 entries, the only inconsistency I see is some entries with a gene count of 0. This may cause some issues later while visualizing, but I have no need to remove those entries right now."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "azdata_cell_guid": "dedc0bfe-17d0-40b2-914f-2ddb54f9ce0d"
   },
   "source": [
    "## Summarize Your Results\n",
    "\n",
    "Make note of your answers to the following questions.\n",
    "\n",
    "1. Did you find all four types of dirty data in your dataset?\n",
    "2. Did the process of cleaning your data give you new insights into your dataset?\n",
    "3. Is there anything you would like to make note of when it comes to manipulating the data and making visualizations?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1. Yes, in some columns there were missing and irregular figures.\n",
    "2. Yes, I am more familiar with the few entities I'm actually interested in explorin"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
